### DISCUSSION TITLE
64 bit APK won't install on Samsung S21 Android 12

### DISCUSSION THREAD

**Discussion by dgurney75** [airsdk/Adobe-Runtime-Support#1706] [General]
Just updated the AIR sdk to 33.1.1.779 and all Distriqt ANes, but get an error on launch with the following message:

APP NAME closed because this app has a bug.

Try updating this app after its developer provides a fix for this error.

It works perfectly fine on Android 11
![02e19b34-c4be-474d-9ceb-04c5c2bd0848](https://user-images.githubusercontent.com/8271407/155414473-681e754b-5ae0-40a1-947e-83064d606da3.jpg)
.

**Comment by FliplineStudios** [comment]
You may want to try targeting SDK 30 (Android 11) in your application XML if you're not already, @marchbold had recommended on the Starling forums to still target SDK 30 if you're using Distriqt ANEs since they haven't been updated to support SDK 31 / Android 12 yet.

**Reply by dgurney75** [reply]
Ok, but i'm already getting reviews from users that have Android 12 complaining. Also, I thought Google said any new updates have to target 31 now.

**Reply by marchbold** [reply]
Current requirements are for Android 30. SDK 31 is expected to become required later in the year (most likely November).

https://support.google.com/googleplay/android-developer/answer/9859152?visit_id=637812522596724383-2975408451&rd=1#targetsdk&zippy=%2Ctarget-api-level-requirements-for-play-console

You should definitely still be using 30 until we have finalised 31 updates for the ANEs. 

There are a lot of changes in 31 and many of the SDKs we use haven't released updates yet.

https://medium.com/airnativeextensions/january-2022-fc59c0861de6

**Reply by ajwfrost** [reply]
Do you have any call stack or logcat information on the crash? or can you send us the APK?

We've seen one issue causing a crash at start-up with the latest SDK but are unable to reproduce that on our devices here so far .. just wondering, if you don't need the new font rendering code (i.e. if you're generally just doing English or latin-style characters) then you could try disabling it with the below line in your application descriptor's `<android>` section:
`<newFontRenderingFromAPI>99999</newFontRenderingFromAPI>`

thanks

**Reply by dgurney75** [reply]
If I target 30, will it run on Android 12 devices? And it looks like it is most likely the ANEs and not the AIR sdk.

**Reply by marchbold** [reply]
yes

**Reply by dgurney75** [reply]
Thanks, Michael.
